dlib/type_safe_union/type_safe_union_kernel_abstract.h

```diff
@@ -345,6 +372,57 @@ namespace dlib
         provides a global swap function
     !*/
 
+// ----------------------------------------------------------------------------------------
+
+    template<
+        typename Variant,
+        typename F
+    >
+    void for_each_type(
+        Variant&& tsu,
+        F&& f
```

Comment:
```suggestion
        F&& f,
        Variant&& tsu
```
How about switching this around to match the order of `visit()`?

Reply:
yep, good idea